test/util/MessagesSpec.scala

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package util
+
+import play.api.i18n.{DefaultMessagesPlugin, Lang, MessagesApi}
+import play.api.test.FakeApplication
+import uk.gov.hmrc.play.test.{UnitSpec, WithFakeApplication}
+
+import scala.util.matching.Regex
+
+class MessagesSpec extends UnitSpec with WithFakeApplication {
+
+  override lazy val fakeApplication = FakeApplication(
+    additionalConfiguration = Map("application.langs" -> "en,cy",
+      "govuk-tax.Test.enableLanguageSwitching" -> true)
+  )
+
+  val messagesAPI: MessagesApi = new DefaultMessagesPlugin(fakeApplication).api
+  val languageEnglish: Lang = Lang.get("en").getOrElse(throw new Exception())
+  val languageWelsh: Lang = Lang.get("cy").getOrElse(throw new Exception())
+  val MatchSingleQuoteOnly: Regex = """\w+'{1}\w+""".r
+  val MatchBacktickQuoteOnly: Regex = """`+""".r
+  val MatchForwardTickQuoteOnly: Regex = """’+""".r
+
+  "Application" should {
+    "have the correct message configs" in {
+      messagesAPI.messages.size shouldBe 4
+      messagesAPI.messages.keys should contain theSameElementsAs Vector("en", "cy", "default", "default.play")
+    }
+
+    "have messages for default and cy only" in {
+      messagesAPI.messages("en").size shouldBe 0
+      val englishMessageCount = messagesAPI.messages("default").size - frameworkProvidedKeys.size
+      messagesAPI.messages("cy").size shouldBe englishMessageCount
+      messagesAPI.messages("default.play").size shouldBe 38
+    }
+  }
+
+  "All message files" should {
+    "have the same set of keys" in {
+      withClue(describeMismatch(defaultMessages.keySet, welshMessages.keySet)) {
+        welshMessages.keySet shouldBe defaultMessages.keySet
+      }
+    }
+    "have a non-empty message for each key" in {
+      assertNonEmptyValuesForDefaultMessages()
+      assertNonEmptyValuesForWelshMessages()
+    }
+    "have no unescaped single quotes in value" in {
+      assertCorrectUseOfQuotesForDefaultMessages()
+      assertCorrectUseOfQuotesForWelshMessages()
+    }
+    "have a resolvable message for keys which take args" in {
```

Comment:
I can't see where this test does what its name implies: check messages for keys?